plugins/postcss-design-tokens/src/data-formats/style-dictionary/v3/value.ts

```diff
@@ -0,0 +1,106 @@
+import valueParser from 'postcss-value-parser';
+
+export type StyleDictionaryV3TokenValue = {
+	cssValue(transformOptions?: TokenTransformOptions): string
+	// The value of the design token. This can be any type of data, a hex string, an integer, a file path to a file, even an object or array.
+	value: unknown
+	// Usually the name for a design token is generated with a name transform, but you can write your own if you choose. By default Style Dictionary will add a default name which is the key of the design token object.
+	name?: string
+	// The comment attribute will show up in a code comment in output files if the format supports it.
+	comment?: string
+	// This is used in formats that support override-able or themeable values like the !default flag in Sass.
+	themeable?: boolean
+	// Extra information about the design token you want to include. Attribute transforms will modify this object so be careful
+	attributes?: unknown
+
+	// Set by the token parser
+	metadata?: {
+		// A default name of the design token that is set to the key of the design token. This is only added if you do not provide one.
+		name?: string
+		// The object path of the design token.
+		path: Array<string>
+		// The file path of the file the token is defined in. This file path is derived from the source or include file path arrays defined in the configuration.
+		filePath: string
+		// If the token is from a file defined in the source array as opposed to include in the configuration.
+		isSource: boolean
+	}
+}
+
+export function extractStyleDictionaryV3Token(node: Record<string, unknown>, key: string, path: Array<string>, filePath: string): StyleDictionaryV3TokenValue {
+	if (typeof node['value'] === 'undefined') {
+		throw new Error('Token value is undefined : ' + path.join('.'));
+	}
+
+	const value = String(node['value']);
+
+	return {
+		value: value,
+		cssValue: (transformOptions?: TokenTransformOptions) => {
+			return applyTransformsToValue(value, transformOptions);
+		},
+		name: String(node['name'] ?? '') || key,
+		comment: String(node['comment'] ?? '') || undefined,
+		metadata: {
+			name: String(node['name'] ?? '') ? key : undefined,
+			path: [...path, key],
+			filePath: filePath,
+			isSource: true,
+		},
+	};
+}
+
+export function applyTransformsToValue(value: string|undefined|null, transformOptions?: TokenTransformOptions): string {
+	if (!value) {
+		return '';
+	}
+
+	if (!transformOptions) {
+		return value;
+	}
+
+	if (transformOptions.toUnit) {
+		const dimension = valueParser.unit(value ?? '');
+		if (!dimension || dimension.unit === transformOptions.toUnit) {
+			return `${value}`;
+		}
+
+		if (dimension.unit === 'rem' && transformOptions.toUnit === 'px') {
+			return remToPx(parseFloat(dimension.number), transformOptions.pluginOptions?.rootFontSize ?? 16);
+		}
+
+		if (dimension.unit === 'px' && transformOptions.toUnit === 'rem') {
+			return pxToRem(parseFloat(dimension.number), transformOptions.pluginOptions?.rootFontSize ?? 16);
```

Comment:
Should we validate & parse `pluginOptions` this would limit the number of checks we need to do down the road and we can offer meaningful errors. 

It's OK to handle later just thinking what happens if someone puts `'foo'` and starts spitting `NaN`, do we want to warn?

Reply:
I added a bunch of type checks around plugin options and value parsing.
This should make the whole more predictable.

> we can offer meaningful errors.

We'll handle this later after we gather some feedback.
For most other plugins we also don't give feedback on incorrect plugin options and we don't receive too many questions about this :)